code/game/objects/items/weapons/grenades/ghettobomb.dm

```diff
@@ -85,6 +117,38 @@
 /obj/item/weapon/grenade/iedcasing/prime() //Blowing that can up
 	update_mob()
 	explosion(get_turf(src.loc),-1,0,2)
+	if(shrapnel_list.len > 0)
+		var/atom/target
+		var/atom/curloc = get_turf(src)
+		var/list/bodyparts = list("head","chest","groin","l_arm","r_arm","l_hand","r_hand","l_leg","r_leg","l_foot","r_foot")
+		for(var/obj/item/shrapnel in shrapnel_list)
+			if(shrapnel.shrapnel_amount >0)
+
+				var/amount = shrapnel.shrapnel_amount
+
+				while(amount > 0)
+					amount--
+					target =pick(trange(6, curloc))
+					var/obj/item/projectile/bullet/shrapnel_projectile = new shrapnel.shrapnel_type(src) //obj/item/projectile/bullet/shrapnel
```

Comment:
is there absolutely no code to just fire a projectile to a turf? This much boilerplate feels unnecessary.